spec_legacy/support/with_mock_request.rb

```diff
@@ -0,0 +1,38 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+RSpec.shared_context 'with mock request for helper', :with_mock_request_for_helper do
+  before do
+    helper.request = ActionController::TestRequest.new(
+      Rack::MockRequest.env_for('/issues',
+        'HTTP_HOST'       => 'test.host',
+        'REMOTE_ADDR'     => '0.0.0.0',
+        'HTTP_USER_AGENT' => 'Rails Testing')
+    )
+  end
+end
```

Comment:
This one seems to be only used in the legacy specs (if my grep foo is correct). So move it rather than copying it.